lib/internal/freelist.js

```diff
@@ -10,7 +10,7 @@ exports.FreeList = function(name, max, constructor) {
 
 
 exports.FreeList.prototype.alloc = function() {
-  return this.list.length ? this.list.shift() :
+  return this.list.length ? this.list.pop() :
                             this.constructor.apply(this, arguments);
```

Comment:
This can also be changed to simply `this.constructor();` for an additional speed increase since `http` is the only module that uses `FreeList` and `http` just passes in a factory function with no parameters as its "constructor."